src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -85,7 +85,8 @@ module Make_update (T : Transaction_snark.Verification.S) = struct
           let%map new_root =
             Pending_coinbase.Checked.add_coinbase root_after_delete
               ( Snark_transition.proposer transition
-              , Snark_transition.coinbase transition )
+              , Snark_transition.coinbase_amount transition
+              , Snark_transition.coinbase_state_body_hash transition )
```

Comment:
There is already a previous state here passed as an argument to this function. Can you use the hash of `body` from that instead of having it in `transition`?

Reply:
It's a `Protocol_state.var`, not a `Protocol_state.value`. Even with an instance of the `value` type, we'd have to compute that hash.

Reply:
You can get `Body.var` and then use `hash_checked` but you're right we don't want to hash twice. But we already compute `Body.hash_checked`  when computing `Protocol_state.hash` in the snark. We could make a function return the body hash along with the state hash. But I don't see the advantage of this over having it in snark transition so I guess it is ok to leave it this way